dashboard/test/controllers/api/v1/user_school_infos_controller_test.rb

```diff
@@ -45,78 +51,549 @@ class UserSchoolInfosControllerTest < ActionDispatch::IntegrationTest
+    assert_nil @teacher.school_info
+    assert_empty @teacher.user_school_infos
   end
 
-  test "school_info_id is updated" do
-    user_school_info = create :user_school_info
-    sign_in user_school_info.user
+  test 'initial, no previoius, partial, manual' do
+    sign_in @teacher
 
-    patch "/api/v1/user_school_infos/#{user_school_info.id}/update_school_info_id", params: {
-      school: {name: 'C school', city: 'Hungtinton river', state: 'Caliii'},
-      school_info: {school_type: 'private', state: 'Jersey New', school_name: 'C school', country: 'US'}
-    }
+    Timecop.travel 1.hour
 
-    new_user_school_info = UserSchoolInfo.last
+    submit_partial_school_info
+    assert_response :success, response.body
 
-    assert_response :success
-    refute_equal new_user_school_info.id, user_school_info.id
-    assert_equal new_user_school_info.school_info_id, user_school_info.school_info_id
+    @teacher.reload
+    refute_nil @teacher.school_info
+    assert @teacher.school_info.school_name.nil?
+    assert_first_tenure(@teacher)
   end
 
-  test "school_info_id is not updated when an invalid school_info id is passed " do
-    user_school_info = create :user_school_info
-    sign_in user_school_info.user
-    patch "/api/v1/user_school_infos/-1/update_school_info_id", params: {
-      school: {name: 'C school', city: 'Hungtinton river', state: 'Caliii'},
-      school_info: {school_type: 'private', state: 'Jersey New', school_name: 'C school', country: 'US'}
+  test 'initial, no previous, complete, drop down' do
+    sign_in @teacher
+
+    school = create :school
+
+    Timecop.travel 1.hour
+
+    submit_complete_school_info_from_dropdown(school)
+
+    @teacher.reload
+
+    assert_response :success, response.body
+
+    refute_nil @teacher.school_info
+    refute_nil @teacher.school_info.school
+    refute_empty @teacher.user_school_infos
+    assert_first_tenure(@teacher)
+  end
+
+  test 'initial, no previous, complete, manual' do
+    sign_in @teacher
+
+    Timecop.travel 1.hour
+
+    assert_creates SchoolInfo do
+      submit_complete_school_info_manual
+    end
+
+    @teacher.reload
+
+    assert_response :success, response.body
+
+    refute_nil @teacher.school_info
+    refute_nil @teacher.school_info.school_name
+    refute_empty @teacher.user_school_infos
+    assert_first_tenure(@teacher)
+  end
+
+  test 'initial, partial previous, blank, manual' do
+    school_info = create :school_info, school_id: nil, validation_type: SchoolInfo::VALIDATION_NONE
+
+    @teacher.update school_info: school_info
+    sign_in @teacher
+
+    refute @teacher.school_info.country.nil?
+
+    Timecop.travel 1.hour
+    submit_blank_school_info
+
+    @teacher.reload
+
+    assert_response :success, response.body
+
+    assert_equal @teacher.school_info.id, school_info.id
+    assert_equal @teacher.school_info, school_info
+    # assert_first_tenure(@teacher)
+    # assert_nil @teacher.school_info.country #Ask Bryan about desired behavior
```

Comment:
Did you get an answer on this?

Reply:
Yes.  Confirmed with Bryan that the school info should not be updated when a blank form is submitted.